addons/mail/models/discuss/discuss_channel.py

```diff
@@ -777,6 +779,7 @@ def _channel_info(self):
             if current_partner or current_guest:
                 info['message_needaction_counter'] = channel.message_needaction_counter
                 member = member_of_current_user_by_channel.get(channel, self.env['discuss.channel.member']).with_prefetch([m.id for m in member_of_current_user_by_channel.values()])
+                info['hasNotSelfAsMember'] = not bool(member)
```

Comment:
I still need a dedicated property to allow server to notifies the client that a specific thread only allow a restricted access to the connected user. I renamed it once again to avoid confusion with the existing "hasSelfAsMember" and still [be closer to the truth](https://github.com/odoo/odoo/pull/137623#discussion_r1379754726)